File: BackAPI/proyecto-node/controllers/mensajeController.js
const MensajeRepository = require('../ado/mensajeRepository');
const mensajeValidator = require('../validators/mensajeValidator');

class MensajeController {
  static async listarMensajesActivos(req, res) {
    try {
      const { error } = mensajeValidator.validate(req.query);
      if (error) return res.status(400).json({ message: error.details[0].message });

      const { mes, clienteId } = req.query;

      const mensajes = await MensajeRepository.listarMensajesActivos(mes, clienteId);
      res.status(200).json(mensajes);
    } catch (err) {
      res.status(500).json({ message: 'Error en el servidor', error: err.message });
    }
  }
}

module.exports = MensajeController;

File: BackAPI/SinapsisAPI/Controllers/MensajeController.cs
using Microsoft.AspNetCore.Mvc;
using SinapsisAPI.Validators;

namespace SinapsisAPI.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class MensajeController : ControllerBase
    {
        private readonly MensajeRepository _repository;

        public MensajeController(IConfiguration configuration)
        {
            string connectionString = configuration.GetConnectionString("DefaultConnection");
            _repository = new MensajeRepository(connectionString);
        }

        // Endpoint para listar mensajes activos filtrados por mes y cliente
        [HttpGet("ListarMensajes")]
        public IActionResult ListarMensajesActivos([FromQuery] int mes, [FromQuery] int? clienteId = null)
        {
            // Validar los parámetros con el validador
            if (!MensajesValidator.ValidarMes(mes))
            {
                return BadRequest("El mes debe estar entre 1 y 12.");
            }

            /*
            if (!MensajesValidator.ValidarClienteId(clienteId))
            {
                return BadRequest("El ID del cliente debe ser mayor a 0.");
            }
            */

            try
            {
                var mensajes = _repository.ListarMensajesActivos(mes, clienteId);
                return Ok(mensajes);
            }
            catch (Exception ex)
            {
                return StatusCode(500, $"Error en el servidor: {ex.Message}");
            }
        }
    }
}
